### PR TITLE
add recipe for sizemeup

### PR DESCRIPTION
**Pull request by rpetit3**
This PR adds a recipe for [sizemeup](https://github.com/rpetit3/sizemeup) a simple tool to get genome sizes based on species names 

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

## Summary by CodeRabbit

- **New Features**
	- Introduced a setup script for easy installation and configuration of the `sizemeup` library.
	- Added a configuration file for `sizemeup`, detailing package information and dependencies.

- **Bug Fixes**
	- Ensured proper permissions and naming for the main executable.

- **Documentation**
	- Included metadata and integrity verification details in the new configuration file.

### CODE REVIEW

**Comment by rpetit3** [comment]
@biocondabot please add label

File: recipes/sizemeup/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  imports:
+    - sizemeup
+  commands:
+    - pip check
+    - sizemeup --version
+    - sizemeup --help
+    - sizemeup --species "Staphylococcus aureus"
+    - sizemeup-build --help
+  requires:
+    - pip
```
[the remark this answers is not in the document]
Comment (rpetit3) [reply recipes/sizemeup/meta.yaml:38]: AI growing pains, `sizemeup-build` does actually exist

File: recipes/sizemeup/build.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# Install sizemup library
+$PYTHON -m pip install --no-deps --ignore-installed -vv .
+
+# move main executable, and replace with wrapper
+chmod 755 bin/sizemeup-bioconda
+mv $PREFIX/bin/sizemeup $PREFIX/bin/sizemeup-main
+cp -f bin/sizemeup-bioconda $PREFIX/bin/sizemeup
+
+# Copy genome size database (~100kb)
+cp -f data/sizemeup-sizes.txt ${PREFIX}/share/sizemeup
```
[the remark this answers is not in the document]
Comment (rpetit3) [reply recipes/sizemeup/build.sh:14]: I'm going to pass on this recommendation as it makes the `build.sh` way more complex then it really needs to be

**Comment by rpetit3** [comment]
@biocondabot please fetch artifacts

**Comment by rpetit3** [comment]
Testing

## Conda

The built in test confirmed conda is working

## Docker
```
docker run --rm -u $(id -u):$(id -g) -v ${PWD}:/data quay.io/biocontainers/sizemeup:1.0.1--py312hdfd78af_0 sizemeup --version
sizemeup-main, version 1.0.1

docker run --rm -u $(id -u):$(id -g) -v ${PWD}:/data quay.io/biocontainers/sizemeup:1.0.1--py312hdfd78af_0 sizemeup --help

 Usage: sizemeup-main [OPTIONS]

 sizemeup - A simple tool to determine the genome size of an organism

╭─ Required Options ───────────────────────────────────────────────────────────╮
│ *  --species  -s  TEXT  The species to determine the size of [required]      │
│ *  --sizes    -z  TEXT  The built in sizes file to use                       │
│                         [default:                                            │
│                         /usr/local/bin/../share/sizemeup/sizemeup-sizes.txt] │
│                         [required]                                           │
╰──────────────────────────────────────────────────────────────────────────────╯
╭─ Additional Options ─────────────────────────────────────────────────────────╮
│ --outdir   -o  PATH  Directory to write output [default: ./]                 │
│ --prefix   -p  TEXT  Prefix to use for output files [default: sizemeup]      │
│ --silent             Only critical errors will be printed                    │
│ --verbose            Increase the verbosity of output                        │
│ --version  -V        Show the version and exit.                              │
│ --help               Show this message and exit.                             │
╰──────────────────────────────────────────────────────────────────────────────╯
docker run --rm -u $(id -u):$(id -g) -v ${PWD}:/data quay.io/biocontainers/sizemeup:1.0.1--py312hdfd78af_0 sizemeup --species "staphylococcus aureus" --outdir /data
PermissionError: [Errno 13] Permission denied: 'sizemeup-sizemeup.txt'
```

Need to fix issue on sizemeup end

**Comment by rpetit3** [comment]
@BiocondaBot please fetch artifacts

**Comment by rpetit3** [comment]
Testing again

```
docker run --rm -u $(id -u):$(id -g) -v ${PWD}:/data quay.io/biocontainers/sizemeup:1.0.2--py312hdfd78af_0 sizemeup --species "staphylococcus aureus" --outdir /data
2024-09-29 22:49:23 INFO     2024-09-29 22:49:23:root:INFO - Reading utils.py:31
                             genome sizes from
                             /usr/local/bin/../share/sizemeup/sizeme
                             up-sizes.txt
                    INFO     2024-09-29 22:49:23:root:INFO -     sizemeup.py:105
                             Found the size of staphylococcus
                             aureus in the sizes file
                    INFO     2024-09-29 22:49:23:root:INFO -     sizemeup.py:106
                             {'name': 'Staphylococcus aureus',
                             'tax_id': '1280', 'size':
                             '2800000', 'source': 'ncbi',
                             'method': 'manually-set'}
                           Query Result
┏━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━┳━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ Name                  ┃ TaxID ┃ Size    ┃ Source ┃ Method       ┃
┡━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━╇━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━━━━━┩
│ staphylococcus aureus │ 1280  │ 2800000 │ ncbi   │ manually-set │
└───────────────────────┴───────┴─────────┴────────┴──────────────┘
Writing the genome size to /data/sizemeup-sizemeup.txt

```

Good to go!

File: recipes/sizemeup/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+requirements:
+  host:
+    - python >=3.12
+    - poetry
+    - pip
+  run:
+    - python >=3.12
+    - requests
+    - rich-click >=1.6.0
```
[the remark this answers is not in the document]
Comment (rpetit3) [reply recipes/sizemeup/meta.yaml]: This one should include poetry

File: recipes/sizemeup/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  imports:
+    - sizemeup
+  commands:
+    - pip check
+    - sizemeup --version
+    - sizemeup --help
+    - sizemeup --species "Staphylococcus aureus"
+    - sizemeup-build --help
+  requires:
+    - pip
```
[the remark this answers is not in the document]
Comment (rpetit3) [reply recipes/sizemeup/meta.yaml:38]: This does not affect the tests and was corrected in the source of sizemeup

**Review by mbhall88 — APPROVED** [review]

**Pull request merged**